scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -1670,13 +1705,22 @@ def _get_containers(
                 if ignore_container is not None and cid == ignore_container:
                     continue
 
-                # Note we need to *include* results that were created after the 'running_or_created_after' time.
-                # that means we need to *ignore* any containers created before that
-                # hence the reason 'create_before' is assigned to a value named '...created_after'
-                if _ignore_old_dead_container(
-                    container, created_before=running_or_created_after
-                ):
-                    continue
+                # AGENT-373/270 - we don't want to ignore containers that are still
+                # waiting to be warmed, otherwise we will miss out on short-lived
+                # logs
```

Comment:
Ok, I would rewrite this a little bit differently in order to better capture what's going on.  I also want to use the `is_old_dead_container`:

```
       is_old_dead_container = _ignore_old_dead_container(container, created_before=running_or_created_after)
       has_been_used = controlled_warmer.has_been_used(container) if controlled_warmer is not None else False
       if is_old_dead_container and has_been_used:
           continue:
```

It kind of would be better to flip the sense of this, but that's hard to do given everything else in this loop just uses continues.  Basically, I want it to be pretty apparent that we allow in alive containers, recently dead containers, and old dead containers if we haven't used their cache entry yet.  Maybe we can list all of those cases just in a comment.

Reply:
Done, but note that I didn't use the ternary syntax `pending_first_use = controlled_warmer.pending_first_use(container) if controlled_warmer is not None else False` because the Black formatter made it just as many lines as the original but made it more awkward to read.